### DISCUSSION TITLE
DigiByte Tor

### DISCUSSION THREAD

**Discussion by ghost** [DigiByte-Core/digibyte#233] [Ideas]
**-----------------------------------------------------------------------------**
**Debian 12: Connect to DigiByte network with Tor.** 
**-----------------------------------------------------------------------------**
1.)     Make file: _install_tor.sh_

```
#!/bin/bash

# Check if running as root
if [ "$(id -u)" -ne 0 ]; then
   echo "This script must be run as root"
   exit 1
fi

# Update the system
apt update && apt install -y apt-transport-https gnupg wget

# Add the Tor Project gpg key
wget -qO- https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc | gpg --dearmor | tee /usr/share/keyrings/tor-archive-keyring.gpg >/dev/null

# Add the Tor Project repository
echo "deb [signed-by=/usr/share/keyrings/torproject-archive-keyring.gpg] https://deb.torproject.org/torproject.org/ bookworm main" | tee /etc/apt/sources.list.d/torproject.list

# Update package list to include new repository
apt update

# Install Tor
apt install -y tor

# Enable and start the Tor service
systemctl enable --now tor

# Check if Tor is running
systemctl status tor --no-pager -l

echo "Tor installation is complete."
```
-----------------------------------------------------------------------------
2.)     `chmod +x install_tor.sh`

-----------------------------------------------------------------------------
3.)     `sudo ./install_tor.sh`

-----------------------------------------------------------------------------
4.)     !!All traffic through tor!!
![tortor](https://github.com/DigiByte-Core/digibyte/assets/125610144/392543e5-8bb3-4394-88fc-858fc6dae9aa)

**-----------------------------------------------------------------------------**
1.)     **Get static tor onion  address**
**-----------------------------------------------------------------------------**

`sudo nano /etc/tor/torrc`

At the end of the torrc file, add the following lines.
```
HiddenServiceDir /var/lib/tor/dgb_onion_service/
HiddenServicePort 12024 127.0.0.1:12024
```
-----------------------------------------------------------------------------
2.)      `sudo systemctl restart tor`

-----------------------------------------------------------------------------
This directory should now contain files.
3.)      `sudo ls /var/lib/tor/dgb_onion_service/`

-----------------------------------------------------------------------------
Your public onion address.
4.)      `sudo cat /var/lib/tor/dgb_onion_service/hostname`

-----------------------------------------------------------------------------
Backup your onion address.
5.)      `sudo cp -r /var/lib/tor/dgb_onion_service/ /path/to/backup_location/`

**Comment by saltedlolly** [comment]
I'm getting a file not found error for this URL: https://deb.torproject.org/torproject.org/apt/torproject-keyring.gpg

**Comment by Sbosvk** [comment]
```
wget -qO- https://deb.torproject.org/torproject.org/A3C4F0F979CAA22CDBA8F512EE8CBC9E886DDD89.asc | gpg --dearmor | tee /usr/share/keyrings/tor-archive-keyring.gpg >/dev/null
```

**Reply by saltedlolly** [reply]
Thanks.

**Reply by Sbosvk** [reply]
Typically, GPG keys are provided in ASCII armored format (hence the dearmor flag), which use the .asc extension, rather than .gpg for the armored key file. It seems like there might be a typo or an outdated link in the URL provided here by Jan.

**Comment by chopperbriano** [comment]
Thanks for typing this update Jan! I have 2 running now. Will add a few more. Appreciate the instructions!

**Comment by saltedlolly** [comment]
In case anyone wants to setup a Tor node, as of the latest release, DigiNode tools now makes it really easy: https://diginode.tools/
